### PR TITLE
New ATM ability

### PR DESCRIPTION
**Pull request by Shad0vvs**
## Why It's Good For The Game

Adds a new ATM easter egg.

## Changelog

:cl:
expansion: Adds a new ATM ability.
soundadd: Added a cat meow sound.
/:cl:

### CODE REVIEW

File: code/modules/economy/ATM.dm [hunk 1]
```diff
@@ -76,9 +76,41 @@ log transactions
 			to_chat(user, SPAN_INFO("You insert [I] into [src]."))
 			src.attack_hand(user)
 			qdel(I)
+	else if(istype(I, /obj/item/holder))
```
Comment (stanalbatross) [inline code/modules/economy/ATM.dm:79]: you probably shouldn't be able to do this with any holder item

subtype the jones holder off a general holder/cat

File: code/modules/economy/ATM.dm [hunk 2]
```diff
@@ -76,9 +76,41 @@ log transactions
...
+		user.visible_message(SPAN_DANGER("[user] begins stuffing [I] into the ATM!"))
+		playsound(src, "sound/machines/fax.ogg", 5)
+		if(!do_after(user, 70, INTERRUPT_ALL, BUSY_ICON_BUILD))
+			return
+		visible_message(SPAN_DANGER("You hear a loud metallic grinding sound."))
+		playsound(src, 'sound/effects/splat.ogg', 25, 1)
+		playsound(src, "sound/effects/bone_break1.ogg", 20)
+		if(istype(I, /obj/item/holder/cat) || istype(I, /obj/item/holder/Jones))
```
Comment (stanalbatross) [inline code/modules/economy/ATM.dm:87]: see above

Comment (stanalbatross) [inline code/modules/economy/ATM.dm:87]: see above

File: code/modules/economy/ATM.dm [hunk 3]
```diff
@@ -76,9 +76,41 @@ log transactions
...
+			playsound(src, "sound/voice/meow.ogg", 15)
```
Comment (stanalbatross) [inline code/modules/economy/ATM.dm:88]: lots of sounds playing at once, consider a delay between these

Comment (stanalbatross) [inline code/modules/economy/ATM.dm:88]: lots of sounds playing at once, consider a delay between these

**Review by stanalbatross — CHANGES_REQUESTED** [review]

File: code/modules/economy/ATM.dm [hunk 4]
```diff
@@ -76,9 +76,39 @@ log transactions
...
+
+		for(var/mob/M in I.contents)
+
+			if(M.client) // if someone was playing the mob, log it
+				M.attack_log += "\[[time_stamp()]\] Was gibbed by <b>[key_name(user)]</b>"
+				user.attack_log += "\[[time_stamp()]\] Gibbed <b>[key_name(M)]</b>"
+				msg_admin_attack("[key_name(user)] gibbed [key_name(M)] in [user.loc.name] ([user.x], [user.y], [user.z]).", user.x, user.y, user.z)
+
+			M.spawn_gibs()
+			M.death(create_cause_data("ATM", user), TRUE)
```
Comment (stanalbatross) [inline code/modules/economy/ATM.dm:96]: might runtime due to cause data being a string instead of datum

Comment (stanalbatross) [inline code/modules/economy/ATM.dm:96]: might runtime due to cause data being a string instead of datum

Comment (Shad0vvs) [reply code/modules/economy/ATM.dm:96]: Did not appear to runtime in testing, please confirm.

Comment (Shad0vvs) [reply code/modules/economy/ATM.dm:96]: Did not appear to runtime in testing, please confirm.

**Review by stanalbatross — APPROVED** [review]

**Review by stanalbatross — APPROVED** [review]

**Pull request merged**